dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -167,24 +185,32 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
                     }
                     else
                     {
+                        auto notificationPlatform{ PushNotificationHelpers::GetNotificationPlatform() };
+
                         // AppId is generated by PushNotificationLongRunningTask singleton
                         wil::unique_cotaskmem_string unpackagedAppUserModelId;
-                        RegisterUnpackagedApplicationHelper(remoteId, unpackagedAppUserModelId);
-                        THROW_IF_FAILED(CreateChannelWithRemoteIdHelper(unpackagedAppUserModelId, remoteId, channelInfo));
+                        THROW_IF_FAILED(notificationPlatform->RegisterFullTrustApplication(m_processName.get(), remoteId, &unpackagedAppUserModelId));
 
-                        auto notificationPlatform{ PushNotificationHelpers::GetNotificationPlatform() };
+                        THROW_IF_FAILED(CreateChannelWithRemoteIdHelper(unpackagedAppUserModelId, remoteId, channelInfo));
 
-                        wil::unique_cotaskmem_string processName;
-                        THROW_IF_FAILED(GetCurrentProcessPath(processName));
-                        THROW_IF_FAILED(notificationPlatform->RegisterLongRunningActivatorWithClsid(processName.get(), GetComServerClsid()));
+                        THROW_IF_FAILED(notificationPlatform->RegisterLongRunningActivatorWithClsid(m_processName.get(), m_registeredClsid));
```

Comment:
no locks for m_processName :(

Reply:
Here and everywhere above

Reply:
https://github.com/microsoft/WindowsAppSDK/pull/2122#discussion_r814358341 

Here the discussion between Daniel and me. What do you think here?

Reply:
While I understand the reasoning behind not locking, as a good habit I would still scope it.

Reply:
Taking an anti-pattern approach should be avoided unless really needed.